consensus/polybft/bridge_manager.go

```diff
@@ -0,0 +1,436 @@
+	consensus.EventTracker
+
+	jsonrpcAddr           string
+	startBlock            uint64
+	stateSenderAddr       types.Address
+	checkpointManagerAddr types.Address
+	exitHelperAddr        types.Address
+	trackerPollInterval   time.Duration
+}
+
+// RelayerState is an interface that defines functions that a relayer store has to implement
+type RelayerState interface {
+	GetAllAvailableRelayerEvents(limit int) (result []*RelayerEventData, err error)
+	UpdateRelayerEvents(events []*RelayerEventData, removeIDs []uint64, dbTx *bolt.Tx) error
+}
+
+// relayerEventsProcessor is a parent struct of both state sync and exit relayer
+// that holds functions common to both relayers
+type relayerEventsProcessor struct {
+	logger     hclog.Logger
+	state      RelayerState
+	blockchain blockchainBackend
+
+	config *relayerConfig
+	sendTx func([]*RelayerEventData) error
+}
+
+// ProcessEvents processes all relayer events that were either successfully or unsuccessfully executed
+// and executes all the events that can be executed in regards to relayerConfig
+func (r relayerEventsProcessor) processEvents() {
+	// we need twice as batch size because events from first batch are possible already sent maxAttemptsToSend times
+	events, err := r.state.GetAllAvailableRelayerEvents(int(r.config.maxEventsPerBatch) * 2)
+	if err != nil {
+		r.logger.Error("retrieving events failed", "err", err)
+
+		return
+	}
+
+	removedEventIDs := []uint64{}
+	sendingEvents := []*RelayerEventData{}
+	currentBlockNumber := r.blockchain.CurrentHeader().Number
+
+	// check already processed events
+	for _, event := range events {
+		// quit if we are still waiting for some old event confirmation (there is no parallelization right now!)
+		if event.SentStatus && event.BlockNumber+r.config.maxBlocksToWaitForResend > currentBlockNumber {
+			return
+		}
+
+		// remove event if it is processed too many times
+		if event.CountTries+1 > r.config.maxAttemptsToSend {
+			removedEventIDs = append(removedEventIDs, event.EventID)
+		} else {
+			event.CountTries++
+			event.BlockNumber = currentBlockNumber
+			event.SentStatus = true
+
+			sendingEvents = append(sendingEvents, event)
+			if len(sendingEvents) == int(r.config.maxEventsPerBatch) {
+				break
+			}
+		}
+	}
+
+	// update state only if needed
+	if len(sendingEvents)+len(removedEventIDs) > 0 {
+		r.logger.Info("sending events", "events", sendingEvents, "removed", removedEventIDs)
+
+		if err := r.state.UpdateRelayerEvents(sendingEvents, removedEventIDs, nil); err != nil {
+			r.logger.Error("updating events failed",
+				"events", sendingEvents, "removed", removedEventIDs, "err", err)
+
+			return
+		}
+	}
+
+	// send tx only if needed
+	if len(sendingEvents) > 0 {
+		if err := r.sendTx(sendingEvents); err != nil {
+			r.logger.Error("failed to send tx", "block", currentBlockNumber, "events", sendingEvents, "err", err)
+		} else {
+			r.logger.Info("tx has been successfully sent", "block", currentBlockNumber, "events", sendingEvents)
```

Comment:
Do we really want to log this at info level? I'd consider maybe lowering to debug and also slightly refactor log message:

```suggestion
			r.logger.Debug("relayer tx has been successfully sent", "block", currentBlockNumber, "events", sendingEvents)
```

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/47/commits/d439dc0c3031276a74d678e4b2b915ca9fd20f74